### PR TITLE
feat: Add statement.bindParameters()

### LINKED ISSUE #34: Add Statement.bindParameters() that accepts an array of values
I purpose to allow  `Statement.bindParam()` to accept an array of values instead of array of arrays like [here]((https://github.com/IBM/nodejs-idb-pconnector/tree/master/docs#statementbindparamparams)).

This should not be difficult we already implement similar behavior for [DBPool.prepareExecute()](https://github.com/IBM/nodejs-idb-pconnector/tree/master/docs#dbpoolprepareexecutesql-params-options).

That way you can use this:

```
  await statement.bindParam([9997, 'Johnson', 'A J', 453 Example','Fort']);
```

rather than this:

```
statement.bindParam([
    [9997, IN, NUMERIC],
    ['Johnson', IN, CHAR],
    ['A J', IN, CHAR],
    ['453 Example', IN, CHAR],
    ['Fort', IN, CHAR],
  ]);
```

### PR DESCRIPTION
**Pull request by abmusse**
Resolves #34

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -174,6 +175,27 @@ Change to system naming and set the library list (using `CHGLIBL`) of the connec
...
 
+## **Statement.bindParameters(params)**
+
+Associates parameter markers in an sql statement to application variables.
+
+**Parameters**:
+
+- **params**: `Array` the parameter list in order corresponding to the parameter markers.
+
+    Each element of the parameter may be literal value of type `string`, `number`, `buffer`, `boolean`, `null`, or `object`.
```
Comment (kadler) [inline docs/README.md]: What does a `boolean` or `object` get mapped to? Db2 has no support for such types.

File: docs/README.md [hunk 2]
```diff
@@ -174,6 +175,27 @@ Change to system naming and set the library list (using `CHGLIBL`) of the connec
...
+    Passing an `object` allows for parameters to be customized at an individual level.
+
+    The object format: 
+    - `value`: `String | Number | Buffer | boolean | null`
+    -  `io`: `in | out | both` default is `both`
+    -  `asClob`: `true | false` default is `false`
```
Comment (kadler) [inline docs/README.md]: Why is `asClob` a thing?

File: docs/README.md [hunk 3]
```diff
@@ -174,6 +175,27 @@ Change to system naming and set the library list (using `CHGLIBL`) of the connec
...
+**Parameters**:
+
+- **params**: `Array` the parameter list in order corresponding to the parameter markers.
+
+    Each element of the parameter may be literal value of type `string`, `number`, `buffer`, `boolean`, `null`, or `object`.
+    Passing an `object` allows for parameters to be customized at an individual level.
+
+    The object format: 
+    - `value`: `String | Number | Buffer | boolean | null`
+    -  `io`: `in | out | both` default is `both`
```
Comment (kadler) [inline docs/README.md]: The driver should know whether they are in/out/both, so why would the user ever need to tell it?

File: docs/README.md [hunk 4]
```diff
@@ -174,6 +175,27 @@ Change to system naming and set the library list (using `CHGLIBL`) of the connec
...
+## **Statement.bindParameters(params)**
+
+Associates parameter markers in an sql statement to application variables.
+
+**Parameters**:
+
+- **params**: `Array` the parameter list in order corresponding to the parameter markers.
+
+    Each element of the parameter may be literal value of type `string`, `number`, `buffer`, `boolean`, `null`, or `object`.
+    Passing an `object` allows for parameters to be customized at an individual level.
```
Comment (kadler) [inline docs/README.md]: Ok, I guess I just needed to read one line down for the object...

**Comment by abmusse** [comment]
> I thought the point of this was to make things simple to bind an array of values.

Agreed. Need to go back to the drawing board on this one.

**Pull request closed without merging**